modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala

```diff
@@ -53,7 +53,11 @@ final case class JavaPreprocessor(
               allowRestrictedFeatures
             )
           }
-          Seq(PreprocessedSource.OnDisk(
+          if (
+            updatedOptions.global == BuildOptions.monoid.zero && updatedOptions.scoped.isEmpty && updatedOptions.unused.isEmpty
```

Comment:
Move this condition to the class `DirectivesProcessorOutput`.

Reply:
Moved checking for no directives to the `ExtractedDirectives` class, I think it makes much more sense there.

Reply:
Good catch!